lua/starfall/libs_sh/players.lua

```diff
@@ -480,39 +531,233 @@ if CLIENT then
+		
+		local anim = playerAnimGet(ply)
+		if not anim then SF.Throw("No animation is playing.", 2) end
+		
+		if isstring(act) then
+			act = ply:LookupSequence(act)
+		elseif act ~= nil and not isnumber(act) then
+			SF.ThrowTypeError("number, string or nil", SF.GetType(act), 2)
+		end
+		
+		anim.activity = act
+	end
+	
+	--- Sets the animation progress
+	-- @client
+	-- @param progress The progress of the animation. Ranging from 0-1
+	function player_methods:setAnimationProgress(progress)
+		local ply = getply(self)
+		checkpermission(instance, ply, "entities.setPlayerRenderProperty")
+		
+		local anim = playerAnimGet(ply)
+		if not anim then SF.Throw("No animation is playing.", 2) end
+		
+		checkluatype(progress, TYPE_NUMBER)
+		
+		anim.progress = progress
+	end
+	
+	--- Sets the animation time
+	-- @client
+	-- @param time The time of the animation in seconds. Float
+	function player_methods:setAnimationTime(time)
+		local ply = getply(self)
+		checkpermission(instance, ply, "entities.setPlayerRenderProperty")
+		
+		local anim = playerAnimGet(ply)
+		if not anim then SF.Throw("No animation is playing.", 2) end
+		
+		checkluatype(time, TYPE_NUMBER)
+		
+		anim.progress = (time / anim.duration - anim.min) * (1 / anim.range)
+	end
+	
+	--- Sets the animation playback rate
+	-- @client
+	-- @param rate The playback rate of the animation. Float
+	function player_methods:setAnimationRate(rate)
+		local ply = getply(self)
+		checkpermission(instance, ply, "entities.setPlayerRenderProperty")
+		
+		local anim = playerAnimGet(ply)
+		if not anim then SF.Throw("No animation is playing.", 2) end
+		
+		checkluatype(rate, TYPE_NUMBER)
+		
+		anim.rate = rate
+	end
+	
+	--- Sets the animation audo advance
+	-- @client
+	-- @param auto_advance Should the animation handle advancing itself. Bool
+	function player_methods:setAnimationAutoAdvance(auto_advance)
+		local ply = getply(self)
+		checkpermission(instance, ply, "entities.setPlayerRenderProperty")
+		
+		local anim = playerAnimGet(ply)
+		if not anim then SF.Throw("No animation is playing.", 2) end
+		
+		anim.auto = auto_advance and true or false
+	end
+	
+	--- Sets the animation bounce
+	-- @client
+	-- @param bounce Should the animation bounce instead of loop. Bool
+	function player_methods:setAnimationBounce(bounce)
+		local ply = getply(self)
+		checkpermission(instance, ply, "entities.setPlayerRenderProperty")
+		
+		local anim = playerAnimGet(ply)
+		if not anim then SF.Throw("No animation is playing.", 2) end
+		
+		anim.bounce = bounce and true or false
+	end
+	
+	--- Sets the animation loop
+	-- @client
+	-- @param loop Should the animation loop. Bool
+	function player_methods:setAnimationLoop(loop)
+		local ply = getply(self)
+		checkpermission(instance, ply, "entities.setPlayerRenderProperty")
+		
+		local anim = playerAnimGet(ply)
+		if not anim then SF.Throw("No animation is playing.", 2) end
+		
+		anim.loop = loop and true or false
```

Comment:
A checktype TYPE_BOOLEAN would be better since users inserting a 0,1 expecting it to work would be confused.